### PR TITLE
Response refactor

### PR DESCRIPTION
**Pull request by keefertaylor**
An initial refactor to remove response handling and parsing out of TezosClient. 

A future refactor will move code that has to do with response parsing out of TezosRPC subclasses and move it to a response handler. Response handler can then be genericized and injected, if customization is desired. 

The end state will be:
- TezosClient - Manages network interface and sending requests
- ResponseHandler - A modular handler on TezosClient that will handle errors
- TezosRPC subclasses - Pure model objects with no business logic

### CODE REVIEW

File: TezosKit/Client/TezosClient.swift [hunk 1]
```diff
@@ -559,41 +563,8 @@ public class TezosClient {
...
-
-        // Drop data and send our error to let subsequent handlers know something went wrong and to
-        // give up.
-        let error = TezosClientError(kind: errorKind, underlyingError: errorMessage)
-        rpc.handleResponse(data: nil, error: error, callbackQueue: self.callbackQueue)
-        return
-      }
-
-      rpc.handleResponse(data: data, error: error, callbackQueue: self.callbackQueue)
+    let request = urlSession.dataTask(with: urlRequest as URLRequest) { data, response, error in
```
Comment (jlaws) [inline TezosKit/Client/TezosClient.swift]: weak self?

File: TezosKit/Client/RPCResponseHandler.swift [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    // Check if the response contained a 200 HTTP OK response. If not, then propagate an error.
+    if let httpResponse = response as? HTTPURLResponse,
+      httpResponse.statusCode != 200 {
+      // Default to unknown error and try to give a more specific error code if it can be narrowed
+      // down based on HTTP response code.
+      var errorKind: TezosClientError.ErrorKind = .unknown
+      // Status code 40X: Bad request was sent to server.
+      if httpResponse.statusCode >= 400, httpResponse.statusCode < 500 {
+        errorKind = .unexpectedRequestFormat
+        // Status code 50X: Bad request was sent to server.
```
Comment (jlaws) [inline TezosKit/Client/RPCResponseHandler.swift]: Bad request was sent to the server for this and 400 errors?

File: TezosKit/Client/RPCResponseHandler.swift [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+  /** Initialize a new response handler with the given callback queue. */
+  public init(callbackQueue: DispatchQueue) {
+    self.callbackQueue = callbackQueue
+  }
+
+  /** Handle the given response from making the given RPC. */
+  public func handleResponse<T>(rpc: TezosRPC<T>, data: Data?, response: URLResponse?, error: Error?) {
+    // Check if the response contained a 200 HTTP OK response. If not, then propagate an error.
+    if let httpResponse = response as? HTTPURLResponse,
```
Comment (jlaws) [inline TezosKit/Client/RPCResponseHandler.swift]: Would response ever be nil in a non-error situation?

File: TezosKit/Client/TezosClient.swift (hunk 1 above)
Comment (keefertaylor) [reply TezosKit/Client/TezosClient.swift]: Done.

File: TezosKit/Client/RPCResponseHandler.swift (hunk 3 above)
Comment (keefertaylor) [reply TezosKit/Client/RPCResponseHandler.swift]: NSURLSession only guarantees a URLResponse object.

What else would you suggest here?

File: TezosKit/Client/RPCResponseHandler.swift (hunk 2 above)
Comment (keefertaylor) [reply TezosKit/Client/RPCResponseHandler.swift]: Error handling could use a bit of love, but I think .unknownResponse is appropriate.

File: TezosKit/Client/RPCResponseHandler.swift (hunk 3 above)
Comment (jlaws) [reply TezosKit/Client/RPCResponseHandler.swift]: Should the argument for this method be ,response: URLResponse, in that case.  Without the optional?

File: TezosKit/Client/RPCResponseHandler.swift (hunk 2 above)
Comment (jlaws) [reply TezosKit/Client/RPCResponseHandler.swift]: I was just questioning the comment, your error seems fine

**Pull request merged**